_posts/en/newsletters/2020-09-30-newsletter.md

```diff
@@ -0,0 +1,189 @@
+  results are possible even if the program's code was well reviewed,
+  strongly tested, formally verified, and otherwise built with the
+  utmost care.  A single incorrect result in the execution of
+  cryptographic or consensus code could result in serious consequences
+  for the users of that code---or anyone who depends on remaining in
+  consensus with users of that code.
+
+    The potential issues affect not only software written in C and
+    compiled with GCC, but any software or library that depends on
+    software or libraries compiled with the affected versions of GCC.
+    That may be a significant amount of software on affected systems,
+    such as recent versions of [Ubuntu and Fedora][maxwell systems].  It
+    includes programs written in other languages whose compilers or
+    interpreters were built using GCC.
+
+    Issues were opened in both the [libsecp256k1][libsecp256k1 #823] and
+    [Bitcoin Core][bitcoin core #20005] repositories to find and
+    mitigate any effects of this bug.  The topic was also
+    [discussed][irc memcmp] during the weekly Bitcoin Core Developers
+    Meeting.
+
+    As of this writing, developers have tested Bitcoin Core 0.20.1 and
+    not found any direct problems.  If any noteworthy problems are found
+    in other software, we'll provide an update in a future newsletter.
+
+- **US Patent 7,110,538 has expired:** Bitcoin transactions are secured using
+  [ECDSA][] (the _Elliptic Curve Digital Signature Algorithm_). Verifying
+  signatures involves multiplying points on the elliptic curve by scalars.
+  Typically, each transaction input requires one or more signature verifications,
+  meaning that syncing the Bitcoin block chain can require many millions of these
+  elliptic curve point multiplications. Any technique to make point
+  multiplications more efficient therefore has the potential to significantly
+  speed up Bitcoin Core's initial sync.
+
+    In a [2011 bitcointalk post][finney endomorphism], Hal Finney described a
+    method by Gallant, Lambert and Vanstone (GLV) to
+    efficiently compute elliptic curve point multiplications using an
+    [endomorphism][] on the curve (a mapping from the curve to itself which
+    preserves all relationships between points). By using this GLV endomorphism,
+    the multiplication can be broken into two parts, which are calculated
+    simultaneously to arrive at the solution. Doing this can reduce the
+    number of expensive computations by up to 33%. Finney wrote a proof-of-concept
+    implementation of the GLV endomorphism, which he claimed sped up signature
+    verification by around 25%.
+
+    Pieter Wuille separately implemented the GLV endomorphism algorithm
+    in the [libsecp][] library, which is used to verify signatures in Bitcoin Core.
+    However, the algorithm was encumbered by [U.S. Patent 7,110,538][endomorphism
+    patent] and so to avoid any legal uncertainty the implementation has not previously been
```

Comment:
s/to avoid any legal uncertainty the implementation/to avoid any legal uncertainty, the implementation